components/coprv2_plugin_api/src/storage_api.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use async_trait::async_trait;
+use std::ops::Range;
+
+/// A raw key in the storage.
+pub type Key = Vec<u8>;
+/// A raw value from the storage.
+pub type Value = Vec<u8>;
+/// A pair of a raw key and its value.
+pub type KvPair = (Key, Value);
+
+/// Result returned by operations on [`RawStorage`].
+pub type StorageResult<T> = std::result::Result<T, StorageError>;
+
+/// Some information about the current region the coprocessor is running in.
+#[derive(Debug, Clone)]
+pub struct Region {
+    pub id: u64,
+    pub start_key: Key,
+    pub end_key: Key,
+    pub region_epoch: RegionEpoch,
+}
+
+#[derive(Debug, Clone)]
+pub struct RegionEpoch {
+    pub conf_ver: u64,
+    pub version: u64,
+}
+
+/// Errors when operating on [`RawStorage`].
+#[derive(Debug)]
+pub enum StorageError {
+    // TODO: this introduces a dependency to kvproto. Is this okay?
```

Comment:
I think it's not ok... Just like `Region`, we have to make reflected types for all the types we need to expose.

Reply:
Okay got it, thanks for the clarification :)

Reply:
I removed the dependency to `kvproto` and just left `KeyNotInRegion` and `OtherError` in the enum. I guess we'll see what other errors we need when I implement the trait for `tikv::storage::Storage`.